### PR TITLE
added 'deploy' command to CLI for deploying a docker image to a container service

### PR DESCRIPTION
**Pull request by bartelsb**

### CODE REVIEW

File: commands/deploy/contract.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	ShortHelp: "Deploy a Docker image to a service.",
+	LongHelp: "`deploy` deploys a Docker image to the given service. " +
+		"For container services, this calls 'docker exec' using the specified image. " +
+		"All other service types cannot be deployed with this command. " +
+		"Here is a sample command\n\n" +
+		"```\ndatica -E \"<your_env_name>\" deploy container01 image01\n```",
+	CmdFunc: func(settings *models.Settings) func(cmd *cli.Cmd) {
+		return func(cmd *cli.Cmd) {
+			serviceName := cmd.StringArg("SERVICE_NAME", "", "The name of the service to deploy to (i.e. 'container01')")
+			imageName := cmd.StringArg("IMAGE_NAME", "", "The name of the image to deploy (i.e. 'image01')")
```
Comment (bhperry) [inline commands/deploy/contract.go]: Small not very important note (but hey I gotta use my 3 years of Latin for something):
i.e. --> id est --> that is --> a clarification of the previous statement
e.g. --> exempli gratia --> for example --> an example of the previous statement

These are exempli gratiae

**Comment by bhperry** [comment]
Just noticed this merge request is for merging into release, but I'm guessing you want to merge into development.

Otherwise, all use of Latin abbreviations are LGTM'd

**Comment by bhperry** [comment]
LGTM

File: commands/deploy/contract.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	"github.com/jault3/mow.cli"
+)
+
+// Cmd is the contract between the user and the CLI. This specifies the command
+// name, arguments, and required/optional arguments and flags for the command.
+var Cmd = models.Command{
+	Name:      "deploy",
+	ShortHelp: "Deploy a Docker image to a service.",
+	LongHelp: "`deploy` deploys a Docker image to the given service. " +
+		"For container services, this calls 'docker exec' using the specified image. " +
```
Comment (nathanjsweet) [inline commands/deploy/contract.go]: Remove this line.

File: commands/deploy/contract.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	"github.com/daticahealth/cli/models"
+	"github.com/jault3/mow.cli"
+)
+
+// Cmd is the contract between the user and the CLI. This specifies the command
+// name, arguments, and required/optional arguments and flags for the command.
+var Cmd = models.Command{
+	Name:      "deploy",
+	ShortHelp: "Deploy a Docker image to a service.",
+	LongHelp: "`deploy` deploys a Docker image to the given service. " +
```
Comment (nathanjsweet) [inline commands/deploy/contract.go]: substitute "for" for "to"

File: commands/deploy/deploy_test.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+		func(w http.ResponseWriter, r *http.Request) {
+			test.AssertEquals(t, r.Method, "GET")
+			fmt.Fprint(w, fmt.Sprintf(`[{"id":"%s","label":"%s","type":"container"}]`, test.SvcID, container))
+		},
+	)
+	mux.HandleFunc("/environments/"+test.EnvID+"/services/"+test.SvcID+"/deploy",
+		func(w http.ResponseWriter, r *http.Request) {
+			test.AssertEquals(t, r.Method, "POST")
+			if strings.Contains(string(r.URL.Query().Get("release")), "/invalid/tag:name") {
+				w.WriteHeader(500)
```
Comment (nathanjsweet) [inline commands/deploy/deploy_test.go]: This should be a 400

**Comment by nathanjsweet** [comment]
LGTM

**Pull request merged**